www/main/navigator/misc/components/DhanGuruPane.jsx

```diff
@@ -0,0 +1,201 @@
+              addDhanGuruSlide(
+                insertSlide.slideStrings.dhanguruStrings[index][
+                  isMiscSlideGurmukhi ? 'gurmukhi' : 'english'
+                ],
+              );
+              setCurrentDhanGuruIndex(index);
+              modal.close();
+              setIsModalOpen(false);
+              modal.destroy();
+            };
+          });
+        };
+
+        // sets the default page to Dhan Guru slide page
+        modal.setContent(slidePage);
+        buttonOnClick();
+        modal.open();
+      }
+    }
+  };
+
+  const getGuruIndex = (index) => {
+    if (index < 9) {
+      return `0${index + 1}`;
+    }
+    return `${index + 1}`;
+  };
+
+  const insertDhanGuru = (index) => {
+    const { english, gurmukhi } = insertSlide.slideStrings.dhanguruStrings[index];
+    setCurrentDhanGuruIndex(index);
+    if (isMiscSlideGurmukhi) {
+      addDhanGuruSlide(gurmukhi);
+    } else {
+      addDhanGuruSlide(english);
+    }
+  };
+
+  const toggleDhanGuruLanguage = (event) => {
+    if (isMiscSlideGurmukhi !== event.target.checked) {
+      setIsMiscSlideGurmukhi(event.target.checked);
+      addDhanGuruSlide(
+        insertSlide.slideStrings.dhanguruStrings[currentDhanGuruIndex][
+          event.target.checked ? 'gurmukhi' : 'english'
+        ],
+      );
+    }
+  };
+
+  useEffect(() => {
+    if (shortcuts.openDhanGuruSlide) {
+      showDhanGuruModal();
+      setShortcuts({
+        ...shortcuts,
+        openDhanGuruSlide: false,
+      });
+    }
+  }, [shortcuts]);
+
+  useEffect(() => {
+    if (isMiscSlide) {
+      ipcRenderer.send('show-misc-text', {
+        text: miscSlideText,
+        isGurmukhi: isMiscSlideGurmukhi,
+        isAnnoucement,
+      });
+    }
+  }, [miscSlideText, isMiscSlide, isMiscSlideGurmukhi, isAnnoucement]);
+
+  return (
+    <div className={className}>
+      <div className="dhan-guru-language-switch">
+        <span>{i18n.t('INSERT.DHAN_GURU_IN_GURMUKHI')}</span>
+        <div className="switch">
+          <input
+            id="dhan-guru-language"
+            name="dhan-guru-language"
+            type="checkbox"
+            checked={isMiscSlideGurmukhi}
+            onChange={toggleDhanGuruLanguage}
+          />
+          <label htmlFor="dhan-guru-language" />
+        </div>
+      </div>
+      <div className="dhan-guru-pane">
+        {gurus.map((guru, index) => (
+          <div
+            className="dhan-guru-button"
+            key={guru}
+            onClick={() => {
+              insertDhanGuru(index);
+            }}
+            onMouseEnter={() => {
+              refsByGuruIndex[guru].current.classList.add('dhan-guru-button-prefix-hover');
+              refsByGuruIndex[guru].current.innerHTML = '';
+            }}
+            onMouseLeave={() => {
+              refsByGuruIndex[guru].current.classList.remove('dhan-guru-button-prefix-hover');
+              refsByGuruIndex[guru].current.innerHTML = getGuruIndex(index);
+            }}
```

Comment:
@theamanjs veerji, I don't think we are using dhan-guru-button-prefix-hover class when styling. It styles directly using hover selector. Is there any other significance of this class?

Also, instead of adding an event here, how about adding an element in span, and setting it to display none on hover?

```
<span className="dhan-guru-button-prefix" ref={refsByGuruIndex[guru]}>
    <span className="prefix-content">  {getGuruIndex(index)} </span>
  </span>
```

In misc.scss, hide prefix-content, when hovered on parent div.

We won't need those events and the references here in that case.

Reply:
I have removed that mouse event code and applied color transparent to the number on hover. You can review that.